Typescript hello world

Reltre opened the pull request:
I had some issues using node version 6.10, whenever I tried to find it on definitely typed it didn't come up as an option. For the time being, I have 6.0.96 specified, this was the latest version specified by definitely typed. 

One other issue I had was using `import` syntax, or at least certain variations of the syntax. I've read that node doesn't entirely support it yet, but I'll keep doing some research in that regard.

Also, I realize that a lot of files were added in this one PR; if you prefer, I can only add the hello world program and some config files in this PR, and maybe add external libraries to another branch or master. Any feedback is much appreciated.

Reltre committed:
Create server that displays "Hello World" in the browser.

Reltre committed:
Change yarn commands to use cached files.

anonmos commented:
Nice!  To answer your points:

> I had some issues using node version 6.10, whenever I tried to find it on definitely typed it didn't come up as an option. For the time being, I have 6.0.96 specified, this was the latest version specified by definitely typed.

That works!  As long as the typings are _roughly_ matching it should work for us.  TypeScript sits on top of it all, so it should translate happily to Javascript that we're able to use, regardless of whether or not the type definition files are exact.

> One other issue I had was using import syntax, or at least certain variations of the syntax. I've read that node doesn't entirely support it yet, but I'll keep doing some research in that regard.

This is part of the beauty behind TypeScript.  You can use `import` syntax, and it should translate to whatever Javascript standard you're targeting.  From your `tsconfig.json` file, it looks like you're targeting `ES5` by default, and the module is `commonjs`.  These should work swimmingly for node, so you can use `import` and it will figure everything out in the translation step.

> Also, I realize that a lot of files were added in this one PR; if you prefer, I can only add the hello world program and some config files in this PR, and maybe add external libraries to another branch or master. Any feedback is much appreciated.

Yep!  Common practice is to add `node_modules` to your `.gitignore`, because other developers should be able to run `yarn install` and rebuild those locally.  Sometimes libraries are built and targeted at the machine that's doing the installing, so it's better that way anyway.  Can you [remove that directory permanently](https://help.github.com/articles/removing-files-from-a-repository-s-history/) for me?

Let me know when you've done that and I'll review the rest!

Reltre committed:
Update versioning for node and typescript in yarn.lock.

Reltre commented:
Ok, good to know. It's strange, even with es2016 and commonjs specified I was getting compilation errors when I tried to use the bracket syntax for module importing. It does work as it is written right now though.

I've removed node_modules from the repo and added it to gitignore.

anonmos commented on .nvmrc:
```diff
@@ -0,0 +1 @@
+6.0.96
```
When I try to run `nvm install` it gives me the following
> Found '/Users/tim/Workspace/Node/fuego-photos/.nvmrc' with version <6.0.96>
Version '6.0.96' not found - try `nvm ls-remote` to browse available versions.

So that's not _actually_ a valid version.  Running `nvm ls-remote` I see:
`v6.0.0
         v6.1.0
         v6.2.0
         v6.2.1
         v6.2.2
         v6.3.0
         v6.3.1
         v6.4.0
         v6.5.0
         v6.6.0
         v6.7.0
         v6.8.0
         v6.8.1
         v6.9.0   (LTS: Boron)
         v6.9.1   (LTS: Boron)
         v6.9.2   (LTS: Boron)
         v6.9.3   (LTS: Boron)
         v6.9.4   (LTS: Boron)
         v6.9.5   (LTS: Boron)
        v6.10.0   (LTS: Boron)
        v6.10.1   (LTS: Boron)
->      v6.10.2   (LTS: Boron)
        v6.10.3   (LTS: Boron)
        v6.11.0   (LTS: Boron)
        v6.11.1   (LTS: Boron)
        v6.11.2   (LTS: Boron)
        v6.11.3   (LTS: Boron)
        v6.11.4   (LTS: Boron)
        v6.11.5   (LTS: Boron)
        v6.12.0   (LTS: Boron)
        v6.12.1   (LTS: Boron)
        v6.12.2   (LTS: Boron)
        v6.12.3   (Latest LTS: Boron)`

We're actually aiming for 6.10.  According to the [DefinitelyTyped/index.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v6/index.d.ts) for this, it looks like it covers all of v6.  So we should be safe to just set it to 6.10 and leave it.

anonmos commented on package.json:
```diff
@@ -0,0 +1,16 @@
+  "author": "",
+  "license": "Unlicense",
+  "private": true,
+  "scripts": {
```
Great!  One thing I've done in the past is add one more script to allow a quick-start by combining the build-and-run into a single script.  Something along the lines of:
`"build:start": "./node_modules/.bin/tsc src/*.ts"`

anonmos reviewed (changes requested):
Nice!  Didn't have to go out of your way to create a web server, but I'm glad you took the extra step and learned about imports.  That will be import-ant later.

Make the `nvm` update for me, then I'll run it and we can merge.

Reltre committed:
Change project node veresion to 6.10.0

Reltre committed:
Update scripts to include build and start command.

Reltre commented:
.nvm file updated and I also added in a build and start command as well. Should be good to go 🚀

anonmos commented:
🔥 👍 LGTM, I'm go for a merge.
